### PR TITLE
EGT support for Proteus (draft)

### PR DESCRIPTION
**Pull request by pchmura4**
Additional changes to enable SPI4/5/6 that looks to be needed to make EGT work for Proteus (draft)

### CODE REVIEW

File: firmware/controllers/algo/rusefi_enums.h [hunk 1]
```diff
@@ -346,6 +346,7 @@ typedef enum __attribute__ ((__packed__)) {
 	SPI_DEVICE_4 = 4,
 	// we seem to have zero SPI5 code
```
[the remark this answers is not in the document]
Comment (pchmura4) [reply firmware/controllers/algo/rusefi_enums.h:347]: sure

File: firmware/hw_layer/ports/stm32/mcuconf_common_f4_f7.h [hunk 2]
```diff
@@ -297,15 +297,15 @@
 #endif
 
 #ifndef STM32_SPI_USE_SPI4
-#define STM32_SPI_USE_SPI4                  FALSE
+#define STM32_SPI_USE_SPI4                  TRUE
```
Comment (rusefillc) [inline firmware/hw_layer/ports/stm32/mcuconf_common_f4_f7.h:300]: danger zone here see how build fails. we are DEFINITELY not enabling all SPI for all boards, please undo mcuconf_common_f4_f7.h change

**Review by rusefillc — CHANGES_REQUESTED** [review]

File: firmware/hw_layer/ports/stm32/mcuconf_common_f4_f7.h (hunk 2 above)
Comment (pchmura4) [reply firmware/hw_layer/ports/stm32/mcuconf_common_f4_f7.h]: done

**Comment by rusefillc** [comment]
@pchmura4 is the following missing?

```
+#if STM32_SPI_USE_SPI5
+	if (spiDevice == SPI_DEVICE_5) {
+		return &SPID5;
+	}
+#endif
+#if STM32_SPI_USE_SPI6
+	if (spiDevice == SPI_DEVICE_6) {
+		return &SPID6;
+	}
 #endif
```

**Comment by pchmura4** [comment]
Added now

**Pull request merged**